compiler/src/symbols.rs

```diff
@@ -0,0 +1,40 @@
+use std::collections::HashMap;
+use whistle_ast::IdentType;
+
+#[derive(Debug, Clone)]
+pub struct SymbolTable {
+  pub scopes: Vec<Scope>,
+  pub types: HashMap<String, IdentType>,
+  pub vars: HashMap<String, Var>,
+}
+
+#[derive(Debug, Clone)]
+pub struct Scope {
+  pub scopes: Vec<Scope>,
+  pub vars: HashMap<String, Var>,
```

Comment:
Does it make sense to split it into both a `vars` and `vals` property for future potential optimizations where we instead of doing a `(local.get $value)` just push the value directly to stack like `(i32.const 123)`. If that even is an optimization, and I guess a tool like `wasm-opt` would do that automatically so might not be needed.